### PR TITLE
Alerting: Fix issue with Slack contact point validation

### LINKED ISSUE #47404: Slack contact point cannot be tested or updated after creating
**What happened**:

Slack contact points can be configured with either recipient+token or with webhook url.

After creating a Slack contact point with either one of these and returning to the edit contact point page, you are not able to test or update without clearing and re-inputting secured inputs as validation will fail.

**What you expected to happen**:

Edit contact form should see that the secure settings are configured when deciding whether required fields are correctly filled in. Thus, letting you test/save the form.

**How to reproduce it (as minimally and precisely as possible)**:

1. Create a new contact point of type Slack.
2. Fill in either webhook or recipient+token (doesn't need to be real values for the purposes of this test)
3. Save the contact point
4. Edit this newly created contact point
5. Attempt to update or test the contact point (validation will fail)

**Anything else we need to know?**:

Peek 2022-04-06 11-45

Hints for PR:
The form requires either token+recipient or webhook and this is accomplished through a `DependsOn` field which dynamically changes the fields' require status based on input. However, the `DependsOn` is attempting to determine a field has a value by looking in `secureSettings`. This will work when initially setting the fields, but as soon as you reload the config `secureSettings` will not be populated, and instead `secureFields` will store a string->bool map if the field is configured.

**Environment**:
- Grafana version: 8.5.0-pre (main)
- OS Grafana is installed on: Ubuntu 20.04
- User OS & Browser: Ubuntu 20.04, Chrome 100

### PR DESCRIPTION
**Pull request by peterholmberg**
**What this PR does / why we need it**:
In https://github.com/grafana/grafana/pull/45618 we added some more validation to support the Slack contact point. This however did not take into account that `secureSettings` is an empty object for existing notifiers. 

**Which issue(s) this PR fixes**:

Fixes #47404

**Special notes for your reviewer**:

### CODE REVIEW

**Review by gillesdemey — APPROVED** [review]
I've helped pair on this so I am biased, but LGTM! 👍  (when CI is green)

**Comment by konrad147** [comment]
I have noticed one weird thing when you try to edit a slack contact point that has token and webhook configured. It's not possible to clear them both
Screen Recording 2022-04-13 at 10 15 35

**Review by alexweav — APPROVED** [review]
LGTM, but I'll defer to experts on the specifics of the frontend validation.

**Pull request merged**
